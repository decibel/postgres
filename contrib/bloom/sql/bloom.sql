CREATE EXTENSION bloom;

CREATE TABLE tst (
	i	int4,
	t	text
);

\copy tst from 'data/data'

CREATE INDEX bloomidx ON tst USING bloom (i, t) WITH (col1 = 3);

SET enable_seqscan=on;
SET enable_bitmapscan=off;
SET enable_indexscan=off;

SELECT count(*) FROM tst WHERE i = 16;
SELECT count(*) FROM tst WHERE t = '5';
SELECT count(*) FROM tst WHERE i = 16 AND t = '5';

SET enable_seqscan=off;
SET enable_bitmapscan=on;
SET enable_indexscan=on;

EXPLAIN (COSTS OFF) SELECT count(*) FROM tst WHERE i = 16;
EXPLAIN (COSTS OFF) SELECT count(*) FROM tst WHERE t = '5';
EXPLAIN (COSTS OFF) SELECT count(*) FROM tst WHERE i = 16 AND t = '5';

SELECT count(*) FROM tst WHERE i = 16;
SELECT count(*) FROM tst WHERE t = '5';
SELECT count(*) FROM tst WHERE i = 16 AND t = '5';

VACUUM ANALYZE tst;

SELECT count(*) FROM tst WHERE i = 16;
SELECT count(*) FROM tst WHERE t = '5';
SELECT count(*) FROM tst WHERE i = 16 AND t = '5';

VACUUM FULL tst;

SELECT count(*) FROM tst WHERE i = 16;
SELECT count(*) FROM tst WHERE t = '5';
SELECT count(*) FROM tst WHERE i = 16 AND t = '5';

RESET enable_seqscan;
RESET enable_bitmapscan;
RESET enable_indexscan;
