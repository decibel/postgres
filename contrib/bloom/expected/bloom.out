CREATE EXTENSION bloom;
CREATE TABLE tst (
	i	int4,
	t	text
);
\copy tst from 'data/data'
CREATE INDEX bloomidx ON tst USING bloom (i, t) WITH (col1 = 3);
SET enable_seqscan=on;
SET enable_bitmapscan=off;
SET enable_indexscan=off;
SELECT count(*) FROM tst WHERE i = 16;
 count 
-------
    14
(1 row)

SELECT count(*) FROM tst WHERE t = '5';
 count 
-------
  1008
(1 row)

SELECT count(*) FROM tst WHERE i = 16 AND t = '5';
 count 
-------
     4
(1 row)

SET enable_seqscan=off;
SET enable_bitmapscan=on;
SET enable_indexscan=on;
EXPLAIN (COSTS OFF) SELECT count(*) FROM tst WHERE i = 16;
                QUERY PLAN                 
-------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on tst
         Recheck Cond: (i = 16)
         ->  Bitmap Index Scan on bloomidx
               Index Cond: (i = 16)
(5 rows)

EXPLAIN (COSTS OFF) SELECT count(*) FROM tst WHERE t = '5';
                QUERY PLAN                 
-------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on tst
         Recheck Cond: (t = '5'::text)
         ->  Bitmap Index Scan on bloomidx
               Index Cond: (t = '5'::text)
(5 rows)

EXPLAIN (COSTS OFF) SELECT count(*) FROM tst WHERE i = 16 AND t = '5';
                        QUERY PLAN                        
----------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on tst
         Recheck Cond: ((i = 16) AND (t = '5'::text))
         ->  Bitmap Index Scan on bloomidx
               Index Cond: ((i = 16) AND (t = '5'::text))
(5 rows)

SELECT count(*) FROM tst WHERE i = 16;
 count 
-------
    14
(1 row)

SELECT count(*) FROM tst WHERE t = '5';
 count 
-------
  1008
(1 row)

SELECT count(*) FROM tst WHERE i = 16 AND t = '5';
 count 
-------
     4
(1 row)

VACUUM ANALYZE tst;
SELECT count(*) FROM tst WHERE i = 16;
 count 
-------
    14
(1 row)

SELECT count(*) FROM tst WHERE t = '5';
 count 
-------
  1008
(1 row)

SELECT count(*) FROM tst WHERE i = 16 AND t = '5';
 count 
-------
     4
(1 row)

VACUUM FULL tst;
SELECT count(*) FROM tst WHERE i = 16;
 count 
-------
    14
(1 row)

SELECT count(*) FROM tst WHERE t = '5';
 count 
-------
  1008
(1 row)

SELECT count(*) FROM tst WHERE i = 16 AND t = '5';
 count 
-------
     4
(1 row)

RESET enable_seqscan;
RESET enable_bitmapscan;
RESET enable_indexscan;
